### PR TITLE
Webspeech

### LINKED ISSUE #56: Integrate google webspeech api
Integrate google webspeech api for :- 
- [ ] Text To Speech
- [ ] Speech input

Yes, I would work on this.

### PR DESCRIPTION
**Pull request by zamhaq**
Fixes #56 

- [x] Text To Speech
- [x] Speech Input

susivoice

There are two commits for two tasks.
 
As you can see from the screenshot, there are is `settings icon` that redirects to a different page to give extension `microphone` access otherwise speech recognition wont work. I have not concentrated much on the UI now. That can be done later. If your speech input does not work you press the mic and speak then first enable `microphone` using `settings icon`.

### CODE REVIEW

**Comment by zamhaq** [comment]
@ms10398 @gabru-md  Should I add `plus plus` to the jshint config. We use it almost all the time.

**Comment by ms10398** [comment]
Yeah sure you can add it.

**Comment by zamhaq** [comment]
@gabru-md @ms10398 Please review.

File: src/script.js [hunk 1]
```diff
@@ -161,6 +176,57 @@ function submitForm() {
...
+//recognition.continuous = true;
+reset();
+recognition.onend = reset();
+
+recognition.onresult = function (event) {
+    var interimText=" ";
+  for (var i = event.resultIndex; i < event.results.length; ++i) {
+    if (event.results[i].isFinal) {
+      textarea.value = event.results[i][0].transcript;
+      console.log(textarea.value);
```
Comment (gabru-md) [inline src/script.js]: We can remove these debugging commands

File: src/script.js [hunk 2]
```diff
@@ -71,6 +77,13 @@ function composeImage(image) {
     return newImg;
 }
 
+function speakOutput(msg){
+    var voiceMsg = new SpeechSynthesisUtterance(msg);
+    console.log(voiceMsg);
+    window.speechSynthesis.speak(voiceMsg);
+    console.log("here speak");
```
Comment (gabru-md) [inline src/script.js]: We can remove this debugging command as wel

**Comment by ms10398** [comment]
Is there any way to check the integration now before merging @zamhaq

**Comment by zamhaq** [comment]
Of course  @ms10398 fetch the pull request locally and test.

**Comment by ms10398** [comment]
Yes i was doing so Thanks

**Comment by ms10398** [comment]
When we click on settings icon it leads to another page can it be that it stays on same page @zamhaq 
And after the first input when we try to give second input using voice it doesn't respond can it be fixed.
We can also go with this merge as a basic functionality then work upon it to make it perfect

**Comment by ms10398** [comment]
@gabru-md What's your thought

**Comment by zamhaq** [comment]
@gabru-md @ms10398 the thing is that you cannot get access to microphone from inside the popup. That must be done from external tab, window but its required only once. So I chose tab. The second thing about click second time works if you wait for a while. That can be improved later. Something is going on with the event that fires. We can work on that later.

**Comment by gabru-md** [comment]
@zamhaq this can be taken in as a good first step towards TTS and STT conversion.
We can definitely improvise this later. Create the known issues that you have. So that we can start working on it. 
On the other hand, it is great. Should be merged so that we can improvise it.
PS : @ms10398 i approve this.

**Pull request merged**